packages/web-bot-auth/src/crypto.ts

```diff
@@ -44,3 +45,138 @@ export class Ed25519Signer {
     return new Uint8Array(signature);
   }
 }
+
+export class RSAPSSSHA512Signer implements Signer {
+  public alg: Algorithm = "rsa-pss-sha512";
+  public keyid: string;
+  private privateKey: CryptoKey;
+
+  constructor(keyid: string, privateKey: CryptoKey) {
+    this.keyid = keyid;
+    this.privateKey = privateKey;
+  }
+
+  static async fromJWK(jwk: JsonWebKey): Promise<RSAPSSSHA512Signer> {
+    const key = await crypto.subtle.importKey(
+      "jwk",
+      jwk,
+      // restricting to RSA-PSS with SHA-512 as other SHA- algorithms are not registered
+      { name: "RSA-PSS", hash: { name: "SHA-512" } },
+      true,
+      ["sign"]
+    );
+    const keyid = await jwkToKeyID(
+      jwk,
+      helpers.WEBCRYPTO_SHA256,
```

Comment:
sha256 or sha512?

Reply:
sha256. This is to compute the keyid, not the signature